src/command_modules/azure-cli-datalake/azure/cli/command_modules/datalake/analytics/custom.py

```diff
@@ -0,0 +1,229 @@
+
+    if not credential_user_password:
+        try:
+            credential_user_password = prompt_pass('Credential Password:')
+        except NoTTYException:
+            # pylint: disable=line-too-long
+            raise CLIError('Please specify both --credential-user-name and --password in non-interactive mode.')
+
+    create_params = DataLakeAnalyticsCatalogCredentialCreateParameters(credential_user_password,
+                                                                       uri,
+                                                                       credential_user_name)
+    client.create_credential(account_name, database_name, credential_name, create_params)
+
+# pylint: disable=too-many-arguments
+def update_adla_catalog_credential(client,
+                                   account_name,
+                                   database_name,
+                                   credential_name,
+                                   credential_user_name,
+                                   uri,
+                                   credential_user_password=None,
+                                   new_credential_user_password=None):
+    if not credential_user_password:
+        try:
+            credential_user_password = prompt_pass('Current Credential Password:')
+        except NoTTYException:
+            # pylint: disable=line-too-long
+            raise CLIError('Please specify --credential-user-name --password and --new-password in non-interactive mode.')
+
+    if not new_credential_user_password:
+        try:
+            new_credential_user_password = prompt_pass('New Credential Password:')
+        except NoTTYException:
+            # pylint: disable=line-too-long
+            raise CLIError('Please specify --credential-user-name --password and --new-password in non-interactive mode.')
+
+    update_params = DataLakeAnalyticsCatalogCredentialUpdateParameters(credential_user_password,
+                                                                       new_credential_user_password,
+                                                                       uri,
+                                                                       credential_user_name)
+    client.update_credential(account_name, database_name, credential_name, update_params)
+
+# job customizations
+# pylint: disable=too-many-arguments
+def submit_adla_job(client,
+                    account_name,
+                    job_name,
+                    script,
+                    runtime_version=None,
+                    compile_mode=None,
+                    compile_only=False,
+                    degree_of_parallelism=1,
+                    priority=1000):
+    contents = None
+    # script can be either script contents or a path to a script
+    # we attempt to always open the script path first, if that fails
+    # then treat it as contents
+    try:
+        with open(script, 'r') as f:
+            contents = f.read()
```

Comment:
@tjprescott I am not familiar with what this would look like, could you give me an example? Also, how would someone use it then? Would they still be able to pass in either a file or an actual script string?